backend/notifications.js

```diff
@@ -0,0 +1,32 @@
+const { PrismaClient } = require('@prisma/client');
+const prisma = new PrismaClient();
+
+const checkNotificationConditions = async (patientId) => {
+  const patient = await prisma.patient.findUnique({
+    where: { id: patientId },
+    include: { appointments: true },
+  });
+
+  if (!patient) {
+    throw new Error('Patient not found');
+  }
+
+  const today = new Date();
+  const age = today.getFullYear() - patient.date_of_birth.getFullYear();
+  const upcomingAppointments = patient.appointments.filter(appointment => {
+    const appointmentDate = new Date(appointment.date);
+    const timeDifference = appointmentDate - today;
+    return timeDifference >= 0 && timeDifference <= 3 * 24 * 60 * 60 * 1000; // 3 days in milliseconds
```

Comment:
It would nice if you could pull out the number 3 and make it a constant variable such as RANGE_OF_DAYS_UPCOMING (feel free to use a better name haha) to make it easier to separate the 3 days logic from the seconds to days conversion, and make it easier to change this number without error in the future